src/main.rs

```diff
@@ -325,16 +330,12 @@ fn decode_and_print_defmt_logs(
 
                 // Forward the defmt frame to our logger.
                 defmt_decoder::log::log_defmt(&frame, file.as_deref(), line, mod_path);
-
-                let num_bytes = buffer.len();
-                buffer.rotate_left(consumed);
-                buffer.truncate(num_bytes - consumed);
             }
 
             Err(defmt_decoder::DecodeError::UnexpectedEof) => break,
 
             Err(defmt_decoder::DecodeError::Malformed) => {
-                log::error!("failed to decode defmt data: {:x?}", buffer);
+                log::error!("failed to decode defmt data");
```

Comment:
by the way, @Urhengulas this looked like the right place to modify to implement #249 (I think)

Reply:
Thank you for the pointer!